datafusion/physical-expr/src/aggregate/average.rs

```diff
@@ -493,104 +569,111 @@ mod tests {
     use super::*;
     use crate::expressions::col;
     use crate::expressions::tests::aggregate;
-    use crate::generic_test_op;
     use arrow::record_batch::RecordBatch;
     use arrow::{array::*, datatypes::*};
-    use datafusion_common::Result;
+    use datafusion_expr::aggregate_function::sum_type_of_avg;
+    use datafusion_expr::type_coercion::aggregates::avg_return_type;
+
+    fn test_with_pre_cast(array: ArrayRef, expected: ScalarValue) {
```

Comment:
This change is necessary because generic_test_op would call Avg::new which would then not have the correct arguments